### DISCUSSION TITLE
Javaagent doesn't support otlphttp exporter

### DISCUSSION THREAD

**Discussion by coder-1111** [open-telemetry/opentelemetry-java-instrumentation#3759] [Q&A]
Hi,

I'm trying to auto instrument my java application. It is built on top of HTTP protocol (not grpc). So we need to configure otlphttp exporter for instrumentation. But I see that otlphttp isn't supported by javaagent. Are you planning to support this exporter or is their any other workaround for this usecase?

**Comment by iNikem** [comment]
Agent's choice of exporter is totally independent from your application's protocols. As long as your destination supports OTLP (e.g. Otel Collector) you are fine.

But to answer your original question: we will add support of OTLP over HTTP when https://github.com/open-telemetry/opentelemetry-java/pull/3418 is merged.

**Reply by coder-1111** [reply]
ok got it. I will use another exporter. Thank you

**Comment by sumitm-iiit** [comment]
@iNikem : It seems that the change to add support of OTLP over HTTP is merged in Opentelemetry/Java. But otlp over http exporter is still not available by default in OpenTelemetry Java Agent. I have tried with Open Telemetry Java Agent 1.5.3 but it seems this change is not present.
Is there a plan to put OTLP over HTTP exporter by default in java agent ?

**Reply by trask** [reply]
hey @sumitm-iiit! I think that support was added to autoconfigure in https://github.com/open-telemetry/opentelemetry-java/pull/3522, which will be in the upcoming Java SDK 1.6.0 and then Java Instrumentation 1.6.0

**Reply by sumitm-iiit** [reply]
@trask : Thanks for your reply :-)

**Reply by trask** [reply]
I realized we also need to pull in the the otlp http exporters too: #4068
